src/proxy/processors/push-action/checkUserPushPermission.ts

```diff
@@ -1,52 +1,31 @@
 import { Action, Step } from '../../actions';
-import { getUsers, isUserPushAllowed } from '../../../db';
-import { trimTrailingDotGit } from '../../../db/helper';
+import { getRepoByUrl, getUsers } from '../../../db';
+
+const getPushUser = async (action: Action): Promise<string | null> => {
+  const list = await getUsers({ gitAccount: action.user });
+  return list.length === 1 ? list[0].username : null;
+};
 
 // Execute if the repo is approved
 const exec = async (req: any, action: Action): Promise<Action> => {
   const step = new Step('checkUserPushPermission');
 
-  const repoSplit = trimTrailingDotGit(action.repo.toLowerCase()).split('/');
```

Comment:
Is the `trimTrailingDotGit` call deferred in the current execution? It seems this is happening implicitly in the `getRepoByUrl` db functions but it's not exactly the same as before:

```ts
  url = url.toLowerCase().replace('.git', '');
```

Perhaps there's a way to modify or reuse the existing `trimTrailingDotGit`?

Reply:
Thank you for letting me know! This makes sense, and I've created an issue to track it: #1113